### ISSUE TITLE
questFinished webhook sent multiple times

### ISSUE THREAD

**Issue by cvuorinen** [HabitRPG/habitica#12036]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### General Info
  * UUID: fa756acf-9127-4a3c-8dac-8ff627d30073

### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)

I have a `questFinished` webhook set up and it used to work just fine, until last Friday (March 27) it started to get called multiple times. Since then it has been called 2-4 times every time we finish a quest.

Expected behavior is to be called only once per quest.

#### Console Errors
[//]: # (Include any JavaScript console errors here.)

**Comment by paglias** [comment]
Hi @cvuorinen and thanks for the report, we pushed some changes to webhooks last week but the main change is better logging on our side and increasing the number of retries. Is the webhook stored in your account? Could you post its id?

**Comment by paglias** [comment]
Looks like this might be partially due to the fact that our webhook sender tries to parse the response as JSON

**Comment by paglias** [comment]
It might be also due to the fact that some of your webhooks seem to return a 302 and our webhook sender isn't currently folllowing 302 redirects https://github.com/sindresorhus/got/issues/568 (we're on version 9, the bug was fixed in 10)

**Comment by cvuorinen** [comment]
The webhook is this one: fadf75a8-2fb1-4211-b09a-b87ab96a14c0

The endpoint is a google script, so I don't there are any redirects. But I just realized that it could take a while to finish, so it's possible it hits some timeout? How long do you wait for response?

**Comment by paglias** [comment]
30 seconds, but ideally it should never be hit. the idea is that you receive the webhook, respond with a success code and then process it

**Comment by paglias** [comment]
Looking at the logs looks like it's a mix of the url returning a 302 code which wasn't being followed correctly (fixed in #12038) and the timeout being hit

**Comment by cvuorinen** [comment]
@paglias yeah, I just looked into it and seems google sometimes returns a 302, just never saw it when testing with Postman as it handled it transparently. Don't know why they do that, but good to know it will get fixed.

Also just now realized the timeout issue, so I'll try to fix it asap.

**Comment by paglias** [comment]
@cvuorinen we've deployed some changes to make sure redirects are followed correctly and non-json responses don't cause any issue, we've also re-enabled any webhook that got disabled in the past few days so hopefully everything should be back to normal, but let me know if it's not

**Comment by cvuorinen** [comment]
@paglias okay, we finished a quest last night and it was called only once :+1:
